app/src/main/java/com/android/streetworkapp/ui/navigation/TopAppBar.kt

```diff
@@ -0,0 +1,37 @@
+package com.android.streetworkapp.ui.navigation
+
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TopAppBarWrapper(navigationActions: NavigationActions, topAppBarManager: TopAppBarManager?) {
+  topAppBarManager?.let {
```

Comment:
Improvement: instead of the nested block you could just add this line and remove the block
```kotlin
if (topAppBarManager == null) return
TopAppBar(...)
```

Reply:
I think it's prettier like this :), I did change `topAppBarManager` to `it` inside the block, was pretty stupid on my part for rereferencing it inside the block